addons/web/static/tests/webclient/navbar_tests.js

```diff
@@ -180,6 +180,34 @@ QUnit.test("navbar can display systray items ordered based on their sequence", a
     navbar.destroy();
 });
 
+QUnit.test("navbar updates after adding a systray item", async (assert) => {
+    class MyItem1 extends Component {}
+    MyItem1.template = xml`<li class="my-item-1">my item 1</li>`;
+
+    clearRegistryWithCleanup(systrayRegistry);
+    systrayRegistry.add('addon.myitem1', { Component: MyItem1 });
+
+    const env = await makeTestEnv(baseConfig);
+    const target = getFixture();
+
+    patchWithCleanup(NavBar.prototype, {
+        mounted() {
```

Comment:
you should patch the `setup` method here, and call the `onMounted` method instead

Reply:
The issue doesn't actually happen when using the `onMounted` callback. I also tried `onWillStart` but no luck.

Reply:
try with onRendered, this way you're before the mounted/patched from useEffect but after it has been rendered with the previous state of the registry.

Reply:
`onRendered` is only available starting from 15.3, no ?

Reply:
Oh right this is 15.0, disregard this comment.